pymeasure/instruments/hp/hp34401A.py

```diff
@@ -22,25 +22,34 @@
 # THE SOFTWARE.
 #
 
+import logging
 from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set
 
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
 
-class HP34401A(Instrument):
-    """ Represents the HP 34401A instrument.
-    """
-
-    voltage_dc = Instrument.measurement("MEAS:VOLT:DC? DEF,DEF", "DC voltage, in Volts")
 
-    voltage_ac = Instrument.measurement("MEAS:VOLT:AC? DEF,DEF", "AC voltage, in Volts")
+def _deprecation_warning(property_name):
+    def func(x):
+        log.warning(f'Deprecated property name "{property_name}", use the "function_" '
+                    'and "reading" properties instead.')
```

Comment:
We really raise a warning (you have to do `from warnings import warn` first), not just a log entry:
```suggestion
        warn((f'Deprecated property name "{property_name}", use the "function_" '
                    'and "reading" properties instead.'), FutureWarning)
```

Reply:
Ok I will do that.